File: fixtures/javascript/type_inference/cross_file/service.js
const { loadUser } = require("./models");

function renderGreeting(user) {
  return user.greet();
}

function main() {
  const user = loadUser("Alice");
  console.log(renderGreeting(user));
  console.log(user.greet());
}

module.exports = { renderGreeting, main };

File: fixtures/go/type_inference/cross_file/service.go
package main

import (
	"fmt"
	"models"
)

func RenderGreeting(user *models.User) string {
	return user.Greet()
}

func main() {
	user := models.LoadUser("Alice")
	fmt.Println(RenderGreeting(user))
	fmt.Println(user.Greet())
}
